File: tmp/copy_files.py
import shutil
import os

def copy_dir(src, dst):
    if os.path.exists(src):
        if not os.path.exists(dst):
            os.makedirs(dst)
        for item in os.listdir(src):
            s = os.path.join(src, item)
            d = os.path.join(dst, item)
            if os.path.isdir(s):
                shutil.copytree(s, d, dirs_exist_ok=True)
            else:
                shutil.copy2(s, d)
        print(f"Copied {src} to {dst}")
    else:
        print(f"Source {src} does not exist")

base = r"c:\Users\Princy Singh\Downloads\allied-surplus"
ref = os.path.join(base, "reference")

copy_dir(os.path.join(ref, "src"), os.path.join(base, "src"))
copy_dir(os.path.join(ref, "app", "_components"), os.path.join(base, "app", "_components"))
copy_dir(os.path.join(ref, "public"), os.path.join(base, "public"))
copy_dir(os.path.join(ref, "redux"), os.path.join(base, "redux"))
shutil.copy2(os.path.join(ref, "data", "products.ts"), os.path.join(base, "data", "products.ts"))
print("Done")

File: tmp/copy_files.js
const fs = require('fs');
const path = require('path');

function copyRecursiveSync(src, dest) {
  const exists = fs.existsSync(src);
  const stats = exists && fs.statSync(src);
  const isDirectory = exists && stats.isDirectory();
  if (isDirectory) {
    if (!fs.existsSync(dest)) {
      fs.mkdirSync(dest, { recursive: true });
    }
    fs.readdirSync(src).forEach((childItemName) => {
      copyRecursiveSync(path.join(src, childItemName), path.join(dest, childItemName));
    });
  } else {
    fs.copyFileSync(src, dest);
  }
}

const base = 'c:\\Users\\Princy Singh\\Downloads\\allied-surplus';
const ref = path.join(base, 'reference');

console.log('Starting copy...');
try {
  copyRecursiveSync(path.join(ref, 'src'), path.join(base, 'src'));
  copyRecursiveSync(path.join(ref, 'app', '_components'), path.join(base, 'app', '_components'));
  copyRecursiveSync(path.join(ref, 'public'), path.join(base, 'public'));
  copyRecursiveSync(path.join(ref, 'redux'), path.join(base, 'redux'));
  fs.copyFileSync(path.join(ref, 'data', 'products.ts'), path.join(base, 'data', 'products.ts'));
  console.log('Copy complete!');
} catch (err) {
  console.error('Error during copy:', err);
}
